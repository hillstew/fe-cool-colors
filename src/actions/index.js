export const setColors = colors => ({
  type: 'SET_COLORS',
  colors
});

export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setProjects = projects => ({
  type: 'SET_PROJECTS',
  projects
});

export const setPalettes = palettes => ({
  type: 'SET_PALETTES',
  palettes
});

export const toggleLocked = id => ({
  type: 'TOGGLE_LOCKED',
  id
});
