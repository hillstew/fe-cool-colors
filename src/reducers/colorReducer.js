export const colorReducer = (state = [], action) => {
  switch (action.type) {
    case 'SET_COLORS':
      return action.colors;
    case 'TOGGLE_LOCKED':
      return [...state].map(color => {
        if (color.id === action.color.id) {
          return { ...color, isLocked: !color.isLocked };
        }
        return color;
      });
    default:
      return state;
  }
};
