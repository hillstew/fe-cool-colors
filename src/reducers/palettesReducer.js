export const palettesReducer = (state = [], action) => {
  switch (action.type) {
    case 'SET_PALETTES':
      return action.palettes;
    case 'SET_PALETTE':
      return [...state, action.palette];
    default:
      return state;
  }
}