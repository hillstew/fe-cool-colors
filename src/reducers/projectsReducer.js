export const projectsReducer = (state = [], action) => {
  switch (action.type) {
    case 'SET_PROJECTS':
      return action.projects;
    case 'SET_PROJECT':
      return [...state, action.project];
    default:
      return state;
  }
};
