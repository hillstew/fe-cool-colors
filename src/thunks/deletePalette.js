import { setLoading, setError, setPalettes } from '../actions';
import { deleteData } from '../utils/api';

export const deletePalette = (id, palettes) => {
  return async dispatch => {
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/palettes/${id}`;
    try {
      dispatch(setLoading(true));
      const response = await deleteData(url);
      const result = await response.json();
      const updatedPalettes = palettes.filter(palette => {
        return palette.id !== id;
      });
      console.log(updatedPalettes);
      dispatch(setLoading(false));
      dispatch(setPalettes(updatedPalettes));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};
