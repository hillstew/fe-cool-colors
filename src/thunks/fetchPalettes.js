import { setError, setLoading, setPalettes } from '../actions';
import { getData } from '../utils/api';

export const fetchPalettes = () => {
  return async dispatch => {
    const url = process.env.REACT_APP_BACKEND_URL + '/api/v1/palettes';
    try {
      dispatch(setLoading(true));
      const response = await getData(url);
      const palettes = await response.json();
      dispatch(setLoading(false));
      dispatch(setPalettes(palettes));
    } catch (error) {
      dispatch(setError(error));
    }
  };
};
