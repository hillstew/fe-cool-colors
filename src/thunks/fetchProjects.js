import { setError, setLoading, setProjects } from '../actions';
import { getData } from '../utils/api';

export const fetchProjects = () => {
  return async dispatch => {
    const url = process.env.REACT_APP_BACKEND_URL + '/api/v1/projects';
    try {
      dispatch(setLoading(true));
      const response = await getData(url);
      const projects = await response.json();
      dispatch(setLoading(false));
      dispatch(setProjects(projects));
    } catch (error) {
      dispatch(setError(error));
    }
  };
};
