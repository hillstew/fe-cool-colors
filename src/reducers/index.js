import { colorReducer } from './colorReducer';
import { errorReducer } from './errorReducer';
import { loadingReducer } from './loadingReducer';
import { projectsReducer } from './projectsReducer';
import { palettesReducer } from './palettesReducer';
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
  color: colorReducer,
  error: errorReducer,
  isLoading: loadingReducer,
  projects: projectsReducer,
  palettes: palettesReducer
});
