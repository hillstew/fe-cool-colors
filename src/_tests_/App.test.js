import React from 'react';
import ReactDOM from 'react-dom';
import {
  App,
  mapStateToProps,
  mapDispatchToProps
} from '../containers/App/App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { rootReducer } from '../reducers';
import { fetchProjects } from '../thunks/fetchProjects';
import { fetchPalettes } from '../thunks/fetchPalettes';
import { setColors } from '../actions';

describe('App', () => {
  const mockProps = {
    fetchProjects: jest.fn(() => true),
    fetchPalettes: jest.fn(() => true),
    setColors: jest.fn()
  };
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = createStore(rootReducer);

    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <App {...mockProps} />
        </BrowserRouter>
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('App component', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<App {...mockProps} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
