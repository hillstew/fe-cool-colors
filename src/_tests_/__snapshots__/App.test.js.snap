// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(ColorContainer) />
  <Connect(ProjectContainer) />
</div>
`;
